Change API documentation location

alecslupu opened the pull request:
#### :tophat: What? Why?
In a call that i have with Andres, we agreed that in the scope of BSC contract to enhance the documentation by adding mutation examples to an API page. 
Apart of that, in this PR: 
- I am adding a link to the Decidim Documentation page
- I am moving the API link from Develop / Advanced / API to Develop / API 
- I am adding a structure that allows us to document any module's mutations (started with the proposals)
 
#### :pushpin: Related Issues
*Link your PR to an issue*
- Related to #?
- Fixes #?

#### Testing
1. Fetch this branch 
2. From documentation repo use the branch of https://github.com/decidim/documentation/pull/249 
3. Build the documentation 

### :camera: Screenshots
*Please add screenshots of the changes you are proposing*
<img width="1254" height="497" alt="image" src="https://github.com/user-attachments/assets/1e025136-a668-4cc4-b102-8da0f7cb3b92" />
<img width="1454" height="860" alt="image" src="https://github.com/user-attachments/assets/9dd692e2-9849-43e7-89e7-841a0d9d3bbf" />
<img width="928" height="342" alt="image" src="https://github.com/user-attachments/assets/5b0af049-755e-4f5e-aaae-013f913b23f1" />

:hearts: Thank you!

alecslupu committed:
Refactor API location

alecslupu committed:
Add Proposal answer mutation example

alecslupu committed:
Add link to decidim docs website

andreslucena reviewed (changes requested):
This isn't a refactor, please change the title. This is adding a new page for mutations

Also we discussed on changing the actual /api/docs to docs.decidim.org - are you planning on doing that in another PR?

I'd prefer to do that to be honest, as I was starting to list all the changes that I'd want, but it probably makes more sense to just do it myself and to have your feedback as reviewer

alecslupu commented:
> This isn't a refactor, please change the title. This is adding a new page for mutations
> 
Changed
> Also we discussed on changing the actual /api/docs to docs.decidim.org - are you planning on doing that in another PR?

This is based on my notes, 
- Add to the api docs the link to docs.decidim examples 
- Move the api link from "advanced" un level up (Develop).

My understanding was that i only need to add some links between `/api/docs` and `docs.decidim.org`. If we want to move the `/api/docs/` to `docs.decidim.org`, we'll need to change the way the documentation is being generated.

Currently the documentation is being generated using a rails gem that we have added back in https://github.com/decidim/decidim/pull/8631. 

Moving it to `docs.decidim.org`, would hide any other documentation that may be added via components or engines. 

> I'd prefer to do that to be honest, as I was starting to list all the changes that I'd want, but it probably makes more sense to just do it myself and to have your feedback as reviewer

Yes, i could be a reviewer.

andreslucena reviewed (approved):
As I mentioned, I have an alternative approach for this.

We'll merge this as is so I can counter-propose my alternative idea :)

andreslucena commented:
Related to #13255
